pennylane/transforms/add_noise.py

```diff
@@ -175,7 +175,7 @@ def circuit(w, x, y, z):
             f"Provided noise model object must define model_map and metatadata attributes, got {noise_model}."
         )
 
-    if level is None or level == "user":  # decompose templates and their adjoints
+    if level is None or level in ("device", "user"):  # decompose templates and their adjoints
```

Comment:
I'm not sure we should add this here.  This does kinda confuse me, but if `level="device"` wasn't suppposed to decompose things before, it shouldn't decompose things after. We already had `level="device"` as a valid option, and in those circumstances, the decomposition didn't apply. 

So I'd say we leave this line as is.

Reply:
Done https://github.com/PennyLaneAI/pennylane/pull/7886/commits/a49fe1392535f212e3e64cfae7664faa17ef99a9